source/compiler/qsc_qasm/src/compiler.rs

```diff
@@ -785,6 +782,24 @@ impl QasmCompiler {
         }
     }
 
+    fn compile_sizeof_call_expr(&mut self, expr: &semast::SizeofCallExpr) -> qsast::Expr {
+        let span = expr.span;
+        let name_span = expr.fn_name_span;
+        let array = self.compile_expr(&expr.array);
+        let dim = self.compile_expr(&expr.dim);
+        let operands = vec![array, dim];
+        let array_dims = expr.array_dims;
+        assert!((1..=7).contains(&array_dims));
```

Comment:
niptick: it's helpful to have some text with the assert.

```suggestion
        assert!((1..=7).contains(&array_dims), "array dimension should be between 1 and 7");
```